### ISSUE TITLE
Positive infinity being returned for Arb.negativeFloat() and vice versa

### ISSUE THREAD

**Issue by CharlesG-Branch** [kotest/kotest#2863]
I'm using `io.kotest:kotest-property:5.0.2` on jvm

Maybe this isn't a bug but it is unexpected. It appears that both positive and negative infinity are being returned in each of the `Arb.negativeFloat()` etc generators.

When I run:
```
println("positiveFloat: ${Arb.positiveFloat().edgecases(10_000)}")
println("negativeFloat: ${Arb.negativeFloat().edgecases(10_000)}")
println("positiveDouble: ${Arb.positiveDouble().edgecases(10_000)}")
println("negativeDouble: ${Arb.negativeDouble().edgecases(10_000)}")
```
This is what I get:
```
positiveFloat: [3.4028235E38, Infinity, 1.0, NaN, 1.4E-45, -Infinity]
negativeFloat: [-3.4028235E38, -1.0, -Infinity, -1.4E-45, NaN, Infinity]
positiveDouble: [Infinity, 1.7976931348623157E308, NaN, 4.9E-324, 1.0, -Infinity]
negativeDouble: [NaN, Infinity, -1.7976931348623157E308, -4.9E-324, -1.0, -Infinity]
```

**Comment by sksamuel** [comment]
Yep that's expected, as mentioned in the edge cases part of the docs.

/**
 * Returns an [Arb] that produces negative [Float]s from [min] to -[Float.MIN_VALUE] (inclusive).
 * The edge cases are [Float.NEGATIVE_INFINITY], [min], -1.0 and -[Float.MIN_VALUE] which are only included if they
 * are in the provided range.
 */

If you don't want to include infinity, then you can use use numericFloat and specify your own bounds.

**Comment by CharlesG-Branch** [comment]
This issue isn't that it includes an infinity, it's that it includes **positive** infinity for `Arb. negativeFloat`. In the docs you just posted it only lists negative infinity not positive infinity as an edgecase.

**Comment by sksamuel** [comment]
Ah I see.

**Comment by sksamuel** [comment]
Will fix for 5.2.2

**Comment by myuwono** [comment]
@sksamuel I'll look into this today
